velox/functions/sparksql/aggregates/FirstLastAggregate.cpp

```diff
@@ -131,8 +138,29 @@ class FirstAggregate : public FirstLastAggregateBase<numeric, TData> {
       bool /* mayPushdown */) override {
     DecodedVector decoded(*args[0], rows);
 
-    rows.applyToSelected(
-        [&](vector_size_t i) { updateValue(i, groups[i], decoded); });
+    rows.applyToSelected([&](vector_size_t i) {
+      updateValue(decoded.index(i), groups[i], decoded.base());
```

Comment:
What is the specific reason for this change, from `decoded` to `decoded.base()`?

Reply:
See line 162, for more cleaner `updateValue` abstraction.

Reply:
OK. It helps to use a generic function, right?

Reply:
Yes